docs/docs/examples/workflow/multi_step_query_engine.ipynb

```diff
@@ -0,0 +1,493 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #106.            nodes = [
</pre><p>tbh if it stop_fn returns true on the first loop, this will be empty. Probably we need to make sure the query engine was run at least once</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/15438/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
Why will the `stop_fn` return true on the first loop?

The `updated_query_bundle` won't be None in the first loop itself, and so the query engine will be run at least once.